e2e/cypress/integration/messaging/emoji_insert_position_spec.js

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// [#] indicates a test step (e.g. # Go to a page)
+// [*] indicates an assertion (e.g. * Check the title)
+// Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Stage: @prod
+// Group: @messaging
+
+describe('Messaging', () => {
+    before(() => {
+    // # Login as "user-1" and go to /
+        cy.apiLogin('user-1');
+        cy.visit('/ad-1/channels/town-square');
+        cy.get('#post_textbox').clear();
+    });
+    it('M23348 - Selecting an emoji from emoji picker should insert it at the cursor position', () => {
+        cy.get('#post_textbox').type('HelloWorld!');
+        cy.get('#post_textbox').type('{leftarrow}{leftarrow}{leftarrow}{leftarrow}{leftarrow}{leftarrow}');
+        cy.get('#emojiPickerButton').click();
+        cy.get('img[data-testid="grinning"]').click();
+        cy.get('#post_textbox').should('have.value', 'Hello :grinning: World!');
+        cy.get('#post_textbox').type('{enter}');
+        cy.getLastPost().find('p').contains('Hello World!');
+        cy.getLastPost().find('p').children();
+        cy.getLastPost().find('p').find('span[data-emoticon="grinning"]').should('be.visible');
```

Comment:
I think this is not accurate to meet "Emoji should be displayed in the post at the position it was inserted". How about something like,
```javascript
cy.getLastPost().find('p').should('have.html', `Hello <span data-emoticon="grinning"><span alt=":grinning:" class="emoticon" title=":grinning:" style="background-image: url(&quot;${Cypress.config('baseUrl')}/static/emoji/1f600.png&quot;);"></span></span> World!`);
```

It's brittle but at least we're sure with the position.

Reply:
I don't 100% like this solution either, but if you said so I'd put it in.

Reply:
Open to suggestion but can't go on with your initial approach.

Reply:
I've tried a few things before opening the PR but could not figure out how to ensure that the grinning emoji is between "Hello" and "World". At least we've tried. :)

Reply:
Thanks for trying. We are working on integrating visual regression testing - https://github.com/mattermost/mattermost-webapp/pull/5256., and this is one of those that fits into it. But on the meantime, we're fine as is.